”make container-serve“ report permission denied

**This is a Bug Report**

run :
```shell
make container-serve
```

output :
```
docker run --rm --interactive --tty --volume /Users/lenovo/workbench/website:/src --mount type=tmpfs,destination=/src/resources,tmpfs-mode=0777 -p 1313:1313 kubernetes-hugo:v0.74.3-9f93d97c7112 hugo server --buildFuture --bind 0.0.0.0
Error: add site dependencies: create deps: failed to create file caches from configuration: mkdir /src/resources/_gen: permission denied
make: *** [container-serve] Error 255
```

my environment:
- mac version：10.14.3 
- bash: 5.0.11
- docker engine：19.03.13

**Proposed Solution:**
I try to figure out what's the problem, and I got following clue:
1. In line 71 of Makefile:
   `--mount type=tmpfs,destination=/src/resources,tmpfs-mode=0777`

2. in container, list directory of /src/resources :
   ```
   drwxr-xr-x    2 root     root            40 Oct  6 15:07 resources/
   ```

3. In line 32 of Dockerfile:
   ```
   USER hugo:hugo
   ```

seems it is a permission problem，
 
 **I simply commented out  lines 32 of dockerfile ` USER hugo:hugo`, and the problem get solved。** 


**Page to Update:**
Makefile
Dockerfile

--- a/Makefile
+++ b/Makefile
@@ -65,10 +65,10 @@ container-image:
 		--build-arg HUGO_VERSION=$(HUGO_VERSION)
 
 container-build: module-check
-	$(CONTAINER_RUN) $(CONTAINER_IMAGE) hugo --minify
+	$(CONTAINER_RUN) --read-only $(CONTAINER_IMAGE) hugo --minify
 
 container-serve: module-check
-	$(CONTAINER_RUN) --mount type=tmpfs,destination=/src/resources,tmpfs-mode=0777 -p 1313:1313 $(CONTAINER_IMAGE) hugo server --buildFuture --bind 0.0.0.0
+	$(CONTAINER_RUN) --read-only --mount type=tmpfs,destination=/tmp,tmpfs-mode=01777 -p 1313:1313 $(CONTAINER_IMAGE) hugo server --buildFuture --bind 0.0.0.0 --destination /tmp/hugo --cleanDestinationDir
 
 test-examples:
 	scripts/test_examples.sh install

--- a/config.toml
+++ b/config.toml
@@ -33,6 +33,23 @@ enableGitInfo = true
 # Hindi is disabled because it's currently in development.
 disableLanguages = ["hi", "no"]
 
+[caches]
+ [caches.assets]
+  dir = ":cacheDir/_gen"
+  maxAge = -1
+ [caches.getcsv]
+  dir = ":cacheDir/:project"
+  maxAge = "60s"
+ [caches.getjson]
+  dir = ":cacheDir/:project"
+  maxAge = "60s"
+ [caches.images]
+  dir = ":cacheDir/_images"
+  maxAge = -1
+ [caches.modules]
+  dir = ":cacheDir/modules"
+  maxAge = -1
+
 [markup]
  [markup.goldmark]
    [markup.goldmark.extensions]

